### PR TITLE
add visibility functions

### PR DESCRIPTION
**Pull request by devomaa**
add visibility functions to window

### CODE REVIEW

File: src/main/resources/data/computercraft/lua/rom/apis/window.lua [hunk 1]
```diff
@@ -235,7 +235,17 @@ function create(parent, nX, nY, nWidth, nHeight, bStartVisible)
     -- @type Window
     -- @see term.Redirect
     local window = {}
-
+    function window.isVisible()
+        return bVisible
+    end
+    
+    function window.setVisible(b)
```
Comment (SquidDev) [inline src/main/resources/data/computercraft/lua/rom/apis/window.lua:242]: There's already a `setVisible` function on line 467. It's probably worth moving `isVisible` to there too.

https://github.com/SquidDev-CC/CC-Tweaked/blob/61fb4caaad51b83ac4a8a5999e50d439f94c71bf/src/main/resources/data/computercraft/lua/rom/apis/window.lua#L467

File: src/main/resources/data/computercraft/lua/rom/apis/window.lua [hunk 2]
```diff
@@ -235,7 +235,17 @@ function create(parent, nX, nY, nWidth, nHeight, bStartVisible)
     -- @type Window
     -- @see term.Redirect
     local window = {}
-
+    function window.isVisible()
```
Comment (SquidDev) [inline src/main/resources/data/computercraft/lua/rom/apis/window.lua:238]: Can you add a brief documentation comment too please? See `getPosition` of the sort of thing you need to say.

File: src/test/resources/test-rom/spec/apis/window_spec.lua [hunk 3]
```diff
@@ -157,4 +157,22 @@ describe("The window library", function()
             expect({ w.getLine(1) }):same { "test ", "aaaa0", "4444f" }
         end)
     end)
+    describe("Window.setVisible", function()
+        it("validates arguments", function()
+            local w = mk()
+            expect.error(w.setVisible, nil):eq("bad argument #1 (expected boolean, got nil)")
+        end)
+        it("sets window visibility", function()
```
Comment (SquidDev) [inline src/test/resources/test-rom/spec/apis/window_spec.lua:165]: I don't think this test is needed, given it's the same as below.

File: src/test/resources/test-rom/spec/apis/window_spec.lua [hunk 4]
```diff
@@ -157,4 +157,22 @@ describe("The window library", function()
...
+           local w = mk()
+           w.setVisible(false)
+           expect(w.isVisible()):same(true)
+        end)
+    end)
+    describe("Window.isVisible", function()
+         it("gets window visibility", function()
+           local w = mk()
+           w.setVisible(false)
+           expect(w.isVisible()):same(true)
```
Comment (SquidDev) [inline src/test/resources/test-rom/spec/apis/window_spec.lua:175]: I don't think you meant to write `true` here. :thinking:

**Review by SquidDev — CHANGES_REQUESTED** [review]
Thanks for the contribution! I've left a couple of comments of a few wee issues I've noticed.

**Review by devomaa — COMMENTED** [review]
fixed it :)

**Comment by SquidDev** [comment]
Ended up merging this manually, sorry! Just had some fixes I wanted to push, and for whatever reason, couldn't update the PR's branch.

**Pull request closed without merging**